### PR TITLE
[poco] Revise netssl, add crypto

### LINKED ISSUE #29629: [poco] Fix handling of feature NetSSL/NetSSLWin
**Is your feature request related to a problem? Please describe.**

Recently there have been some modifications of the port (6a57c0e10fa8eafc58f3ff5ce110c82b5a167908) to change the way the modules NetSSL and NetSSLWin are handled. To understand the made changes, it is essential to understand that on Windows you may build either NetSSL or NetSSLWin, but not both. NetSSLWin is not supported on another platform than Windows.

However the way the port works at the moment, it has the following disadvantages:
- It is not possible to use the NetSSL (not NetSSLWin) module on platform Windows. The port always uses NetSSLWin on platform Windows.
- It is not possible to use the Crypto module at all on platform Windows. The port always disables the module Crypto on platform Windows.
- It is not possible to use the Crypto module without using NetSSL on all other platforms. The port enables the module Crypto only and only, if the module NetSSL is enabled (on all platforms other than Windows).

**Proposed solution**
A solution is to declare NetSSLWin instead of NetSSL as a feature (supported on platform Windows only). If that feature is enabled/requested on platform Windows, then NetSSLWin is build _instead_ of NetSSL. Otherwise NetSSL is build on platform Windows. On all other platforms NetSSLWin is never enabled, but NetSSL is. So the proposed solution fulfills the requirement that either NetSSL or NetSSLWin may be enabled on Windows and never enabled NetSSLWin on platforms other than Windows.

Additionally the coupling of the Crypto module to NetSSL/NetSSLWin has to be removed, so that it is build always no matter, if NetSSL/NetSSLWin is build.

### PR DESCRIPTION
**Pull request by dg0yt**
Revises the netssl handling introduced in #27582.
Resolves #29629. This is an alternative to #29630. See this PR for full background. CC @i-C-o-d-e-r.

- Adds a feature crypto. This is a module which needs openssl and which is required for (OpenSSL) NetSSL.
- Changes the netssl feature to use the (OpenSSL) NetSSL module.
  The NetSSL_Win module is an *alternative* API which is not officially released. So it is not a good default, and it cannot be offered as a feature. However:
- Add and document a variable which can be set in the triplet file in order to activate NetSSL_Win instead of (OpenSSL) NetSSL.
  (Note that this variable will not disable the dependency on openssl / feature "crypto".)

---

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [X] The "supports" clause reflects platforms that may be fixed by this new version
- [X] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [X] Any patches that are no longer applied are deleted from the port's directory.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Review by i-C-o-d-e-r — DISMISSED** [review]

File: ports/poco/portfile.cmake [hunk 1]
```diff
@@ -58,14 +58,11 @@ vcpkg_cmake_configure(
         -DPOCO_MT=${POCO_MT}
         -DENABLE_TESTS=OFF
         # Allow enabling and disabling components
-        # POCO_ENABLE_SQL_ODBC, POCO_ENABLE_SQL_MYSQL and POCO_ENABLE_SQL_POSTGRESQL are
```
Comment (i-C-o-d-e-r) [inline ports/poco/portfile.cmake:67]: The removal of outdated comments is unrelated to what you want to achieve. So should not be part of this PR IMO.

**Comment by i-C-o-d-e-r** [comment]
I think issue #29629 should be mentioned here as this PR is an alternative to the PR #29630, which tries to solve that issue. So merging this pull request may close this issue.

File: ports/poco/portfile.cmake (hunk 1 above)
Comment (dg0yt) [reply ports/poco/portfile.cmake:67]: Outdated comments are a burden for future maintenance, and each new revision forces rebuilds on users. So there is no reason to wait with this trivial change. For review purposes, the PR is structured into separate commits.

**Review by JonLiu1993 — APPROVED** [review]

**Comment by JonLiu1993** [comment]
Feature `crypto` tested successfully in the following triplet:
- x86-windows
- x64-windows
- x64-windows-static
- x64-linux

**Review by i-C-o-d-e-r — APPROVED** [review]

File: ports/poco/portfile.cmake [hunk 2]
```diff
@@ -28,24 +28,19 @@ string(COMPARE EQUAL "${VCPKG_CRT_LINKAGE}" "static" POCO_MT)
...
-            -DENABLE_NETSSL=ON
-            -DENABLE_CRYPTO=ON
-        )
-    endif()
+# POCO_ENABLE_NETSSL_WIN: 
+# Use the unreleased NetSSL_Win module instead of (OpenSSL) NetSSL.
+# This is a variable which can be set in the triplet file.
+if(POCO_ENABLE_NETSSL_WIN)
+    string(REPLACE "ENABLE_NETSSL" "ENABLE_NETSSL_WIN" FEATURE_OPTIONS "${FEATURE_OPTIONS}")
+    list(APPEND FEATURE_OPTIONS "-DENABLE_NETSSL:BOOL=OFF")
```
Comment (JavierMatosD) [inline ports/poco/portfile.cmake:43]: Why do we need this?

Comment (dg0yt) [reply ports/poco/portfile.cmake:43]: Why do we need `"-DENABLE_NETSSL:BOOL=OFF"`?
The answer is in the past discussions and bugs.
- We must not use both `ENABLE_NETSSL_WIN` and `ENABLE_NETSSL` at the same time, because incompatible headers of one module overwrite the headers of the other.
- POCOs build system doesn't take care of this conflict.

I want to use `vcpkg_check_featured` for the common case. This spot is to ensure clear `FEATURE_OPTIONS` for the exceptional case (triplet switch).

**Review by dan-shaw — APPROVED** [review]
